### ISSUE TITLE
[AWS-Sagemekar] TensorBoardOutputConfig

### ISSUE THREAD

**Issue by adityakillekar** [aws/amazon-sagemaker-examples#3620]
I'm using sagemaker (2.111.0) PyTorch estimator. I have configured the tensorboard log directory as follows:

**from sagemaker.debugger import TensorBoardOutputConfig
tensorboard_output_config = TensorBoardOutputConfig(
    s3_output_path=f"s3://<bucket-name>",
    container_local_output_path='/opt/ml/checkpoints/tb'
)**

I'm getting the following error:
**botocore.exceptions.ClientError: An error occurred (ValidationException) when calling the CreateTrainingJob operation: "LocalPath" of "TensorBoardOutputConfig" cannot start with the following reserved path: [/opt/ml, /tmp, /usr/local/nvidia]**

I have configured it like this before but it is not allowing me to use /opt/ml now. And if I don't do so the tensorboard logs won't be copied to my s3 bucket in real-time, rendering tensorboard useless.

The reason for having tensorboard logs in real-time is to save cost if my training job is going sideways. Now there is no way to know if my job is running as expected until the job is finished.

Any help is much appreciated!

Best regards,
Aditya

**Comment by lukasugar** [comment]
Same happening to me, I'm using version `2.153.0`

**Comment by maltelorbach** [comment]
Still the same for 2.173.0. Does anyone know how it's supposed to work? It doesn't even accept the default value for `LocalPath`:

> **LocalPath**
> Path to local storage location for tensorBoard output. Defaults to /opt/ml/output/tensorboard.
Source: https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_TensorBoardOutputConfig.html

**Comment by maltelorbach** [comment]
> I've hacked it like this (copying only relevant stuff). Snippet from my job submission script:
> 
> ```
> DEFAULT_LOCAL_TENSORBOARD_PATH = '/opt/ml/output/tensorboard'
> 
> # Tensorboard config
> tensorboard_output_config = TensorBoardOutputConfig(
>     s3_output_path=f's3://{bucket_name}/training/{job_name}/tensorboard',
>     # Using default container_local_output_path = DEFAULT_LOCAL_TENSORBOARD_PATH
> )
> 
> # Create an estimator
> estimator = PyTorch(entry_point='training_script.py',
>                     hyperparameters={    
>                         'logging_dir': DEFAULT_LOCAL_TENSORBOARD_PATH,
>                     },
>                     tensorboard_output_config = tensorboard_output_config,
> )
> ```

Right, then I suppose your `training_script.py` deals with writing the tensorboard files to the correct location. Too bad we can't change it.

Thanks!

**Comment by adityakillekar** [comment]
@lukasugar @maltelorbach here "correct location" being the s3 path or somewhere else in local instance?

**Comment by lukasugar** [comment]
> @lukasugar @maltelorbach here "correct location" being the s3 path or somewhere else in local instance?

The "correct location" is the `logging_dir` that's being passed as hyperparameter.

In this case, that's `DEFAULT_LOCAL_TENSORBOARD_PATH`. Because of the TensorboardConfig, everything written in this location is getting copied to S3.

**Comment by nbrosse** [comment]
Same issue here, would be glad to have a fix on this
